native/spark-expr/src/cast.rs

```diff
@@ -723,7 +728,9 @@ fn cast_array(
             timezone,
             allow_incompat,
         )?),
-        _ if is_datafusion_spark_compatible(from_type, to_type, allow_incompat) => {
+        _ if ugly_hack_for_poc
```

Comment:
What are the cases (that we know of)  where this gets invoked? (If we know we can replace this flag with an explicit check for those cases, perhaps?)

Reply:
I believe that we need to implement specific logic for adapting parquet schemas, rather than re-using our Spark-compatible cast. There is likely some overlap, so we can refactor the common code out. For example, regular spark casts do not need to support unsigned integers, but we need this when adapting Parquet schemas.